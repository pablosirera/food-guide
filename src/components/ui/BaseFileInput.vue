<script setup>
import { ref } from 'vue'

defineProps({
  label: {
    type: String,
    default: '',
  },
  accept: {
    type: String,
    default: 'image/*',
  },
})
const emit = defineEmits(['select-file'])

const fileName = ref('')

const selectFile = target => {
  fileName.value = target.files[0].name
  emit('select-file', target.files[0])
}
</script>

<template>
  <div class="form-control w-full max-w-xs">
    <span v-if="fileName">{{ fileName }}</span>
    <button
      type="button"
      class="btn btn-md btn-accent btn-outline"
      @click="$refs.file.click()"
    >
      {{ label }}
    </button>
    <input
      ref="file"
      type="file"
      :accept="accept"
      class="hidden"
      @input="selectFile($event.target)"
    />
  </div>
</template>
