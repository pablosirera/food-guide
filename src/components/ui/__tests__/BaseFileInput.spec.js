import { mount } from '@vue/test-utils'
import BaseFileInput from '@/components/ui/BaseFileInput.vue'
import { describe, expect, it } from 'vitest'

describe('Components > UI > BaseFileInput', () => {
  it('should be render', () => {
    const wrapper = mount(BaseFileInput)
    expect(wrapper.exists).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
