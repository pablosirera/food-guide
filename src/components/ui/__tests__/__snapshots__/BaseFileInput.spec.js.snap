// Vitest Snapshot v1

exports[`Components > UI > BaseFileInput > should be render 1`] = `
"<div class=\\"form-control w-full max-w-xs\\">
  <!--v-if--><button type=\\"button\\" class=\\"btn btn-md btn-accent btn-outline\\"></button><input type=\\"file\\" accept=\\"image/*\\" class=\\"hidden\\" />
</div>"
`;
